vmaas/reposcan/katello.py

```diff
@@ -0,0 +1,184 @@
+KATELLO_ORG_LIST_API = "/katello/api/v2/organizations"
+KATELLO_REPOS_LIST_API = "/katello/api/v2/repositories?organization_id=%s&search=redhat=true"
+
+RETRY_COUNT = int(os.getenv("RETRY_COUNT", "3"))
+
+
+class KatelloApiException(Exception):
+    """Error happened during Katello API usage."""
+
+
+class KatelloApi:
+    """Class used to obtain information from Satellite/Katello API."""
+    def __init__(self, hostname: str, api_user: str, api_pass: str):
+        self.hostname: str = hostname
+        self.api_user: str = api_user
+        self.api_pass: str = api_pass
+
+        self.session = Session()
+        retries = Retry(
+            total=RETRY_COUNT,
+            backoff_factor=2,
+        )
+        self.session.mount('http://', HTTPAdapter(max_retries=retries))
+        self.session.mount('https://', HTTPAdapter(max_retries=retries))
+
+        self.tmp_directory: str | None = None
+
+        repo_store: RepositoryStore = RepositoryStore()
+        self.archs: list[str] = repo_store.list_archs()
+
+    def _rm_tmpdir(self) -> None:
+        if self.tmp_directory:
+            LOGGER.debug("Deleting tmp dir: %s", self.tmp_directory)
+            shutil.rmtree(self.tmp_directory)
+        self.tmp_directory = None
+
+    def _find_arch(self, url: str) -> str | None:
+        for arch in self.archs:
+            if arch in url:
+                return arch
+        return None
+
+    def _fetch_katello(  # pylint: disable=too-many-branches
+        self, endpoint, scheme="https", json=True, timeout=30, **kwargs
+    ) -> dict:
+        """Sends request to Katello server/API"""
+
+        if scheme == "https":
+            verify = os.path.join(self.tmp_directory, os.path.basename(KATELLO_CA_CERT_PATH))
+            auth = (self.api_user, self.api_pass)
+        else:
+            verify = True
+            auth = None
+
+        if json:
+            headers = {"Accept": "application/json"}
+        else:
+            headers = {}
+
+        url = f"{scheme}://{self.hostname}{endpoint}"
+
+        try:
+            response = self.session.request(
+                method="GET",
+                url=url,
+                headers=headers,
+                timeout=timeout,
+                auth=auth,
+                verify=verify,
+                **kwargs,
+            )
+            if response.status_code == 200:
+                if json:
+                    return response.json()
+                return {"data": response.text}
+
+            LOGGER.error("Error during GET request to url %s: HTTP %s, %s", url, response.status_code, response.text)
+        except RequestException:
+            LOGGER.exception("Error calling API %s: ", url)
+
+        return {}
+
+    def _download_katello_ca_certificate(self) -> str:
+        LOGGER.info("Downloading Katello CA certificate.")
+
+        ca_certificate = self._fetch_katello(KATELLO_CA_CERT_PATH, scheme="http", json=False)
```

Comment:
I'm not sure if this is a good idea to download the CA cert over http. Can we have this CA certificate feed by a secret or have it mounted as a volume (or on a certain path in the container)?

Reply:
Let's do it later